packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/DatasetResource.java

```diff
@@ -183,4 +214,87 @@ public Response uploadFile(
 	) {
 		return proxy.uploadFile(id, filename, file);
 	}
+	
+	// TODO: https://github.com/DARPA-ASKEM/Terarium/issues/1005
+	// warning this is not sufficient for the long term. Should likely use a library for this conversion formatting may break this.
+	private List<List<String>> csvToRecords(String rawCsvString){
+		String[] csvRows = rawCsvString.split("\n");
+		String[] headers = csvRows[0].split(",");
+		List<List<String>> csv = new ArrayList<List<String>>();
+		for (int i = 0; i < csvRows.length; i++){
+			csv.add(Arrays.asList(csvRows[i].split(",")));
+		}
+		return csv;
+	}
+	
+	private List<String> getColumn(List<List<String>> maxtrix, int columnNumber){
+		List<String> column = new ArrayList<>();
+		for(int i = 0 ; i < maxtrix.size(); i++){
+		   column.add(i,maxtrix.get(i).get(columnNumber));
+		}
+		return column;
+	}
+
+	//Given a column and an amount of bins create a CsvColumnStats object.
+	private CsvColumnStats getStats(List<String> aCol, Integer binCount){
+		List<Integer> bins = new ArrayList<>();
+		try {
+			// set up row as numbers. may fail here.
+			// List<Integer> numberList = aCol.stream().map(String s -> Integer.parseInt(s.trim()));
+			List<Double> numberList = aCol.stream().map(Double::valueOf).collect(Collectors.toList());
+			Collections.sort(numberList); 
+			double minValue = numberList.get(0);
+			double maxValue = numberList.get(numberList.size() - 1);
+			double meanValue = numberList.stream()
+				.mapToDouble(d -> d)
+				.average()
+				.orElse(0.0);
+
+			double medianValue = getMedianValue(numberList);
+			double sdValue = getSdValue(numberList, meanValue);
+			//Set up bins
+			for (int i = 0; i < binCount; i++){
+				bins.add(0);
+			}	
+			double stepSize = (numberList.get(numberList.size() - 1) - numberList.get(0)) / (binCount - 1);
+			
+			// Fill bins:
+			for (int i = 0; i < numberList.size(); i++){
+				Integer index = (int)Math.abs(Math.floor((numberList.get(i) - numberList.get(0)) / stepSize));
+				Integer value = bins.get(index);
+				bins.set(index,value + 1);
+			}
+			
+			return new CsvColumnStats(bins,minValue,maxValue,meanValue,medianValue,sdValue);
+		
+		}catch(RuntimeException e){
```

Comment:
The `RuntimeException` is mostly being thrown by our proxy handler error catching, which I don't see a call to proxy here. You'll likely want a different exception type, or just use `Exception` to catch anything